pluggy.py

```diff
@@ -689,8 +677,8 @@ def varnames(func):
 
     args, defaults = spec.args, spec.defaults
     index = -len(defaults) if defaults else None
-    args, defaults = args[slice(0, index)], args[slice(
-        index, None if index else 0)]
+    args = args[:index]
```

Comment:
that leads to always empty defaults due to replacing args and is till no if/else without a slice all